lib/elixir/lib/module.ex

```diff
@@ -749,11 +765,15 @@ defmodule Module do
       end
 
   """
-  def defines?(module, tuple, kind) do
+  @spec defines?(module, function_arity, def_kind) :: boolean
+  def defines?(module, {function_macro_name, arity} = tuple, def_kind)
+      when is_atom(module) and is_atom(function_macro_name) and
+           is_integer(arity) and arity >= 0 and arity <= 255 and
+           def_kind in [:def, :defp, :defmacro, :defmacrop] do
     assert_not_compiled!(:defines?, module)
     table = defs_table_for(module)
     case :ets.lookup(table, {:def, tuple}) do
-      [{_, ^kind, _, _, _, _}] -> true
+      [{_, ^def_kind, _, _, _, _}] -> true
```

Comment:
this is done intentionally, to be inlined with the `typep`s:
in add_doc is called `kind` because its typespec is a union `def_kind | type_kind`